lms/djangoapps/grades/new/course_data.py

```diff
@@ -20,23 +25,17 @@ def __init__(self, user, course=None, collected_block_structure=None, course_str
     @property
     def course_key(self):
         if not self._course_key:
-            if self._course:
-                self._course_key = self._course.id
-            else:
-                structure = self._structure or self._collected_block_structure
-                self._course_key = structure.root_block_usage_key.course_key
+            self._course_key = self.course.id
         return self._course_key
 
     @property
     def location(self):
         if not self._location:
-            structure = self._structure or self._collected_block_structure
+            structure = self.effective_structure
             if structure:
                 self._location = structure.root_block_usage_key
-            elif self._course:
-                self._location = self._course.location
             else:
-                self._location = modulestore().make_course_usage_key(self._course_key)
+                self._location = self.course.location
```

Comment:
Same here - no reason to access Mongo/instantiate course/etc just to convert the course-key to its course-location.  I would want to revert this change as well.